spec/ruby/core/kernel/trap_spec.rb

```diff
@@ -1,12 +1,11 @@
 require_relative '../../spec_helper'
 require_relative 'fixtures/classes'
+require_relative '../../shared/signal/trap'
 
 describe "Kernel#trap" do
   it "is a private method" do
     Kernel.should have_private_instance_method(:trap)
   end
-end
 
-describe "Kernel.trap" do
-  it "needs to be reviewed for spec completeness"
+  it_behaves_like :signal_trap, "Kernel", KernelSpecs::Method.new
```

Comment:
I try to avoid the approach of sharing specs for slow specs like these.
`Kernel#trap` is just delegating to `Signal.trap` and it doesn't seem worth to run twice as many specs for it (and also having to handle tags for both).
Same thing for Kernel#spawn and Process.spawn for example, we have detailed specs for one of them, but there is little point to run all specs again on the other.

Could you move the specs back in `spec/ruby/core/signal/trap_spec.rb`?

Reply:
Do you think we need anything in `kernel/trap_spec.rb`? A test that the private method exists and then just a comment saying tested elsewhere? Currently says 'needs to be reviewed'.

Reply:
> A test that the private method exists and then just a comment saying tested elsewhere?

Yes, that sounds good.